pkg/handlers/index/alias_es.go

```diff
@@ -0,0 +1,184 @@
+	Actions []Action `json:"actions"`
+}
+
+type Action struct {
+	Add    *base `json:"add"`
+	Remove *base `json:"remove"`
+}
+
+type base struct {
+	Index   string   `json:"index"`
+	Alias   string   `json:"alias"`
+	Indices []string `json:"indices"`
+	Aliases []string `json:"aliases"`
+}
+
+// @Id AddOrRemoveESAlias
+// @Summary Add or remove index alias for compatible ES
+// @Tags    Index
+// @Produce json
+// @Success 200 {object} map[string]interface{}
+// @Failure 400 {object} meta.HTTPResponseError
+// @Router /es/_aliases [post]
+func AddOrRemoveESAlias(c *gin.Context) {
+	var alias Alias
+	err := zutils.GinBindJSON(c, &alias)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, meta.HTTPResponseError{Error: err.Error()})
+		return
+	}
+
+	addMap := map[string][]string{}
+	removeMap := map[string][]string{}
+
+	indexList := core.ZINC_INDEX_LIST.List()
+
+outerLoop:
+	for _, action := range alias.Actions {
+		if action.Add != nil {
+			if action.Add.Index != "" {
+				matchAndAddToMap(indexList, action.Add.Index, addMap, action.Add)
+				continue outerLoop
+			}
+
+			// index is empty, try the indices field
+			for _, indexName := range action.Add.Indices {
+				matchAndAddToMap(indexList, indexName, addMap, action.Add)
+			}
+
+			continue outerLoop // this was an add action, don't bother checking action.Remove
+		}
+
+		if action.Remove != nil {
+			if action.Remove.Index != "" {
+				matchAndAddToMap(indexList, action.Remove.Index, removeMap, action.Remove)
+				continue outerLoop
+			}
+
+			// index is empty, try the indices field
+			for _, indexName := range action.Remove.Indices {
+				matchAndAddToMap(indexList, indexName, removeMap, action.Remove)
+			}
+		}
+	}
+
+	for alias, indexes := range addMap {
+		_ = core.ZINC_INDEX_ALIAS_LIST.AddIndexesToAlias(alias, indexes)
+	}
+
+	for alias, indexes := range removeMap {
+		_ = core.ZINC_INDEX_ALIAS_LIST.RemoveIndexesFromAlias(alias, indexes)
+	}
+
+	c.JSON(http.StatusOK, gin.H{"acknowledged": true})
+}
+
+// @Id GetESAliases
+// @Summary Get index alias for compatible ES
+// @Tags    Index
+// @Produce json
+// @Param   index path  string  false  "Index"
+// @Param   alias path  string  false  "Alias"
+// @Success 200 {object} map[string]interface{}
+// @Failure 400 {object} meta.HTTPResponseError
+// @Router /es/{target}/_alias/{target_alias} [get]
+func GetESAliases(c *gin.Context) {
+	targetIndex := c.Param("target")
+
+	var targetIndexes []string
+	if targetIndex != "" {
+		targetIndexes = strings.Split(targetIndex, ",")
+	}
+
+	targetAlias := c.Param("target_alias")
+
+	var targetAliases []string
+	if targetAlias != "" {
+		targetAliases = strings.Split(targetAlias, ",")
+	}
+
+	m := core.ZINC_INDEX_ALIAS_LIST.GetAliasMap(targetIndexes, targetAliases)
+
+	c.JSON(http.StatusOK, m)
+}
+
+func indexNameMatches(name, indexName string) bool {
+	if name == indexName {
+		return true
+	}
+
+	if strings.Contains(name, "*") {
+		p, err := getRegex(name)
+		if err != nil {
```

Comment:
About wildcard match, i remember we already have it:
https://github.com/zinclabs/zinc/blob/main/pkg/handlers/search/search_v2.go#L153

Reply:
Yes, but this just helps to encapsulate that logic for alias

Reply:
I mean, there is no need do like this complex regex. just prefix or suffix can work well.

Reply:
There is no possiblity of `log*-2022`?